src/modules/dashboard/components/ActionsPage/MultisigWidget/MultisigWidget.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactNode, useState } from 'react';
+import { ColonyRole } from '@colony/colony-js';
+import { nanoid } from 'nanoid';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Address } from '~types/index';
+import styles from './MultisigWidget.css';
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import PermissionsLabel from '~core/PermissionsLabel';
+import ProgressBar from '~core/ProgressBar';
+
+const displayName = 'dashboard.ActionsPage.MultisigWidget';
+
+const MSG = defineMessages({
+  title: {
+    id: 'ActionsPage.MultisigWidget.title',
+    defaultMessage: `Members with {permissionLabel} permission`,
+  },
+  progressBarLabel: {
+    id: 'ActionsPage.MultisigWidget.progressBarLabel',
+    defaultMessage: `{value} of {max} signatures ({required} required)`,
+  },
+});
+
+const UserAvatar = HookedUserAvatar({ fetchUser: true });
+
+interface Props {
+  // List of user addresses which are allowed to approve
+  membersAllowedForApproval: Address[];
+  // Required number of approval
+  requiredNumber: number;
+  /* Required perimission. For now we have only case with recorvery mode.
+  But in future we can have other usecases */
+  requiredPermission: ColonyRole;
+  // Children
+  children?: ReactNode;
+}
+
+const MultisigWidget = ({
+  membersAllowedForApproval,
+  requiredNumber,
+  requiredPermission,
+  children,
+}: Props) => {
+  const [autogeneratedIds] = useState<string[]>(
+    [...new Array(membersAllowedForApproval.length)].map(nanoid),
+  );
```

Comment:
Nice. I see you noticed the thing I did in `IndexModal` :)

Reply:
of course i did :D